[Code health] Reenable codecov PR comments

When uploading the report to Codecov we see the following in GCB logs:

```
Already have image (with digest): gcr.io/ground-android-gcb/android:base
[2024-07-30T19:22:09.973Z] ['verbose'] Start of uploader: 1722367329973...
[2024-07-30T19:22:09.976Z] ['info'] 
     _____          _
    / ____|        | |
   | |     ___   __| | ___  ___ _____   __
   | |    / _ \ / _` |/ _ \/ __/ _ \ \ / /
   | |___| (_) | (_| |  __/ (_| (_) \ V /
    \_____\___/ \__,_|\___|\___\___/ \_/

  Codecov report uploader 0.7.3
[2024-07-30T19:22:10.016Z] ['info'] => Project root located at: /workspace
[2024-07-30T19:22:10.022Z] ['info'] ->  Token found by arguments
[2024-07-30T19:22:10.024Z] ['verbose'] Start of network processing...
[2024-07-30T19:22:10.024Z] ['verbose'] Searching for files in /workspace
[2024-07-30T19:22:10.042Z] ['verbose'] coveragepy is not installed
[2024-07-30T19:22:10.042Z] ['info'] Searching for coverage files...
[2024-07-30T19:22:12.590Z] ['verbose'] Preparing to clean the following coverage paths: coverage-service.xml
[2024-07-30T19:22:12.590Z] ['error'] None of the following appear to exist as files: coverage-service.xml
[2024-07-30T19:22:12.591Z] ['error'] There was an error running the uploader: Error while cleaning paths. No paths matched existing files!
[2024-07-30T19:22:12.591Z] ['verbose'] The error stack is: Error: Error while cleaning paths. No paths matched existing files!
    at cleanCoverageFilePaths (/snapshot/repo/dist/src/helpers/files.js)
    at main (/snapshot/repo/dist/src/index.js)
[2024-07-30T19:22:12.592Z] ['verbose'] End of uploader: 2619 milliseconds
[2024-07-30T19:22:12.592Z] ['info'] Codecov will exit with status code 0. If you are expecting a non-zero exit code, please pass in the `-Z` flag
```

`coverage-service.xml` is the wrong path; we need to set the path to wherever JaCoCo writes the report.

Also note that `testDevDebugUnitTest` and therefore also `jacocoTestStagingUnitTestReport` fail with:

```
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':ground:generateDevDebugProto'.
> protoc: stdout: . stderr: com/google/android/ground/proto/AuditInfoKt.kt: Tried to write the same file twice.
  com/google/android/ground/proto/AuditInfoOuterClassKt.kt: Tried to write the same file twice.
  com/google/android/ground/proto/GeometryKt.kt: Tried to write the same file twice.
  com/google/android/ground/proto/CoordinatesKt.kt: Tried to write the same file twice.
  com/google/android/ground/proto/PointKt.kt: Tried to write the same file twice.
  com/google/android/ground/proto/LinearRingKt.kt: Tried to write the same file twice.
...
```

--- a/build.gradle
+++ b/build.gradle
@@ -38,7 +38,7 @@ buildscript {
         classpath "androidx.navigation:navigation-safe-args-gradle-plugin:$navigationVersion"
         classpath 'com.android.tools.build:gradle:8.3.2'
         classpath 'com.google.gms:google-services:4.4.1'
-        classpath "com.pascalwelsch.gitversioner:gitversioner:0.5.0"
+        classpath "com.pascalwelsch.gitversioner:gitversioner:0.6.3"
 
         // Performance Monitoring plugin: https://firebase.google.com/docs/perf-mon
         classpath 'com.google.firebase:perf-plugin:1.4.2'

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -203,7 +203,7 @@ steps:
       - |
         curl -Os https://uploader.codecov.io/latest/linux/codecov
         chmod +x codecov
-        ./codecov --verbose upload-process --fail-on-error -t ${_CODECOV_TOKEN} -n 'service'-$BUILD_ID -F service -f coverage-service.xml
+        ./codecov --verbose upload-process --fail-on-error -t ${_CODECOV_TOKEN} -n 'service'-$BUILD_ID -F service -f ground/build/reports/jacoco/jacocoDevDebugUnitTestReport/jacocoDevDebugUnitTestReport.xml
 
   - name: 'gcr.io/$PROJECT_ID/android:base'
     id: &compress_cache 'Compress gradle build cache'

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -113,9 +113,7 @@ steps:
         ./gradlew -PdisablePreDex testDevStagingUnitTest --no-daemon 2> unit-test-logs.txt || echo "fail" > build-status.txt
         cat unit-test-logs.txt
 
-        if [[ "${_PUSH_TO_MASTER}" ]]; then
-          ./gradlew jacocoDevDebugUnitTestReport --no-daemon
-        fi        
+        ./gradlew jacocoDevDebugUnitTestReport --no-daemon
 
   - name: 'gcr.io/$PROJECT_ID/android:34'
     id: &authenticate_gcloud 'Authorize gcloud'

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -114,7 +114,7 @@ steps:
         cat unit-test-logs.txt
 
         if [[ "${_PUSH_TO_MASTER}" ]]; then
-          ./gradlew jacocoTestStagingUnitTestReport --no-daemon
+          ./gradlew jacocoDevDebugUnitTestReport --no-daemon
         fi        
 
   - name: 'gcr.io/$PROJECT_ID/android:34'

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -110,7 +110,7 @@ steps:
     args:
       - '-c'
       - |
-        ./gradlew -PdisablePreDex testDevStagingUnitTest --no-daemon 2> unit-test-logs.txt || echo "fail" > build-status.txt
+        ./gradlew -PdisablePreDex testDevDebugUnitTest --no-daemon 2> unit-test-logs.txt || echo "fail" > build-status.txt
         cat unit-test-logs.txt
 
         ./gradlew jacocoDevDebugUnitTestReport --no-daemon

--- a/config/jacoco/jacoco.gradle
+++ b/config/jacoco/jacoco.gradle
@@ -25,7 +25,7 @@ tasks.withType(Test).configureEach {
     jacoco.excludes = ['jdk.internal.*']
 }
 
-tasks.register('jacocoTestStagingUnitTestReport', JacocoReport) {
+tasks.register('jacocoDevDebugUnitTestReport', JacocoReport) {
     dependsOn 'testDevDebugUnitTest'
     group = "Reporting"
     description = "Run tests and generate coverage reports"

